File: app/src/main/java/com/github/kongpf8848/animation/views/AnimationRadioView.java
package com.github.kongpf8848.animation.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.Checkable;

import com.airbnb.lottie.LottieAnimationView;

public class AnimationRadioView extends LottieAnimationView implements Checkable {

    private boolean checked;

    public AnimationRadioView(Context context) {
        this(context, null);
    }

    public AnimationRadioView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public AnimationRadioView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    @Override
    public boolean isChecked() {
        return this.checked;
    }

    @Override
    public void setChecked(boolean checked) {

        try {
            if (this.checked != checked) {
                this.checked = checked;
                if (isAnimating()) {
                    cancelAnimation();
                }
                if (checked) {
                    if (getSpeed() < 0.0F) {
                        reverseAnimationSpeed();
                    }
                    playAnimation();
                } else {
                    if (getSpeed() > 0.0F) {
                        reverseAnimationSpeed();
                    }
                    playAnimation();
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void toggle() {
        setChecked(!this.checked);
    }
}

File: app/src/main/java/com/github/kongpf8848/animation/views/AnimationRadioView.kt
package com.github.kongpf8848.animation.views

import android.content.Context
import android.util.AttributeSet
import android.widget.Checkable
import com.airbnb.lottie.LottieAnimationView

class AnimationRadioView @JvmOverloads constructor(context: Context?, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : LottieAnimationView(context, attrs, defStyleAttr), Checkable {

    private var checked = false

    override fun isChecked(): Boolean {
        return checked
    }

    override fun setChecked(checked: Boolean) {
        try {
            if (this.checked != checked) {
                this.checked = checked
                if (isAnimating) {
                    cancelAnimation()
                }
                if (checked) {
                    if (speed < 0.0f) {
                        reverseAnimationSpeed()
                    }
                    playAnimation()
                } else {
                    if (speed > 0.0f) {
                        reverseAnimationSpeed()
                    }
                    playAnimation()
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun toggle() {
        isChecked = !checked
    }
}